guacamole/src/main/frontend/src/app/client/types/ManagedClientGroup.js

```diff
@@ -0,0 +1,359 @@
+     *     ManagedClientGroup.
+     */
+    ManagedClientGroup.getClientIdentifiers = function getClientIdentifiers(group) {
+
+        if (_.isString(group))
+            return group.split(/\./);
+
+        return group.clients.map(client => client.id);
+
+    };
+
+    /**
+     * Returns the number of columns that should be used to evenly arrange
+     * all provided clients in a tiled grid.
+     *
+     * @returns {Number}
+     *     The number of columns that should be used for the grid of
+     *     clients.
+     */
+    ManagedClientGroup.getColumns = function getColumns(group) {
+
+        if (!group.clients.length)
+            return 0;
+
+        return Math.ceil(Math.sqrt(group.clients.length));
+
+    };
+
+    /**
+     * Returns the number of rows that should be used to evenly arrange all
+     * provided clients in a tiled grid.
+     *
+     * @returns {Number}
+     *     The number of rows that should be used for the grid of clients.
+     */
+    ManagedClientGroup.getRows = function getRows(group) {
+
+        if (!group.clients.length)
+            return 0;
+
+        return Math.ceil(group.clients.length / ManagedClientGroup.getColumns(group));
+
+    };
+
+    /**
+     * Returns the title which should be displayed as the page title if the
+     * given client group is attached to the interface.
+     *
+     * @param {ManagedClientGroup} group
+     *     The ManagedClientGroup to determine the title of.
+     *
+     * @returns {string}
+     *     The title of the given ManagedClientGroup.
+     */
+    ManagedClientGroup.getTitle = function getTitle(group) {
+
+        // Use client-specific title if only one client
+        if (group.clients.length === 1)
+            return group.clients[0].title;
+
+        // With multiple clients, somehow combining multiple page titles would
+        // be confusing. Instead, use the combined names.
+        return ManagedClientGroup.getName(group);
+
+    };
+
+    /**
+     * Returns the combined names of all clients within the given
+     * ManagedClientGroup, as determined by the names of the associated
+     * connections or connection groups.
+     *
+     * @param {ManagedClientGroup} group
+     *     The ManagedClientGroup to determine the name of.
+     *
+     * @returns {string}
+     *     The combined names of all clients within the given
+     *     ManagedClientGroup.
+     */
+    ManagedClientGroup.getName = function getName(group) {
+
+        // Generate a name from ONLY the focused clients, unless there are no
+        // focused clients
+        var relevantClients = _.filter(group.clients, client => client.clientProperties.focused);
+        if (!relevantClients.length)
+            relevantClients = group.clients;
+
+        return _.filter(relevantClients, (client => !!client.name)).map(client => client.name).join(', ') || '...';
+
+    };
+
+    /**
+     * A callback that is invoked for a ManagedClient within a ManagedClientGroup.
```

Comment:
What callback is this referring to?

Reply:
I guess it's referring to the callback invoked by `forEach` below?

Reply:
Yes. It's type documentation for the callback used by `forEach`.